### ISSUE TITLE
User Model

### ISSUE THREAD

**Issue by Kashish0423** [Girl-Code-It/Opportunity-Calendar-Backend#37]
I think we should create a User Model for different users , so that they can view various opportunities and apply to them as per their need and requirements , along with this a moderator who can  verify if the job posted is genuine or not

**Comment by Manvityagi** [comment]
Yes, We need a user model. There will be two types of users - 
- visitors on the website: Can view and post opportunties, The posted opportunities will not be directly posted on the portal
- admins: All the rights (CRUD) along with the ability to verify the opportunities posted by visitors. 
     - For this, we need a whole new admin-panel. 

These are my suggestions, We can discuss the pros and cons of it and other approaches that maybe used. 
Basically a feature-decision. 

CC: @abdus

**Comment by Kashish0423** [comment]
Yes , we can go ahead with the plan of having these 2 users as you mentioned ,
 but I feel we can break the Visitor in two categories  
 
 1. The person who is looking for Job Opportunity(Internship, FTE, Freelancing) 
 2. The person who is posting a Job(i.e HR, Recruiter, Organisation)

Then it will be more easy to manage the one who is looking for Job and the one who is posting the Job (so the database will be 
 more specific to a group of people and easy to maintain ) along with a

3.  moderators(admins) as you rightly mentioned .

In any case whatever suits well we can go ahead with the plan of things asap 
and yes I am comfortable in MongoDB (so we can go ahead with the plan)

**Comment by Manvityagi** [comment]
> Yes , we can go ahead with the plan of having these 2 users as you mentioned ,
> but I feel we can break the Visitor in two categories
> 
> 1. The person who is looking for Job Opportunity(Internship, FTE, Freelancing)
> 2. The person who is posting a Job(i.e HR, Recruiter, Organisation)
> 
> Then it will be more easy to manage the one who is looking for Job and the one who is posting the Job (so the database will be
> more specific to a group of people and easy to maintain ) along with a
> 
> 1. moderators(admins) as you rightly mentioned .
> 
> In any case whatever suits well we can go ahead with the plan of things asap
> and yes I am comfortable in MongoDB (so we can go ahead with the plan)

```
The person who is looking for Job Opportunity(Internship, FTE, Freelancing)
The person who is posting a Job(i.e HR, Recruiter, Organisation)
```
About these, Opportunity Calendar is not really looking to invite Recruiters (they can come and post jobs, but that's not the basic idea of the project), 
The project aims to enable opportunity sharing with easy to browse and share features, as in people like you and me only will share opportunities on the portal, not necessarily for our own company, but for any companies/organizations at all, 

Putting in a simpler way - The idea is to build a social media of opportunities.
So, as of now, We won't have recruiters and users separately. 

Let's say you got to know about some scholarship, you are not the scholarship provider, but to help others know about it you can still post it on Opportunity Calendar. 


CC: @DeepanshiD @vaishali614 @abdus

**Comment by Manvityagi** [comment]
Let's have `user` and `admin` models in this issue. 
Only models, no interactions with the APIs. @Kashish0423

**Comment by Kashish0423** [comment]
Okay this works, I get it , so shall I first present the scheme of things that I will be implementing(based on the things I understand ) or else you assign me the issue and see how it works and suggest changes accordingly so that it fits  well with the demand of the project

**Comment by Manvityagi** [comment]
> Okay this works, I get it , so shall I first present the scheme of things that I will be implementing(based on the things I understand ) or else you assign me the issue and see how it works and suggest changes accordingly so that it fits well with the demand of the project


Assigned to you, second option looks good.
